crates/goose-server/src/routes/config_management.rs

```diff
@@ -206,100 +222,123 @@ pub async fn add_extension(
-    if extensions.remove(&query.key).is_some() {
-        // Save updated extensions
-        match config.set_param(
-            "extensions",
-            Value::Object(extensions.into_iter().collect()),
-        ) {
-            Ok(_) => Ok(Json(format!("Removed extension {}", query.key))),
-            Err(_) => Err(StatusCode::INTERNAL_SERVER_ERROR),
-        }
-    } else {
-        Err(StatusCode::NOT_FOUND)
+    let key = name_to_key(&name);
+    // Use ExtensionManager to remove the extension
+    match ExtensionManager::remove(&key) {
+        Ok(_) => Ok(Json(format!("Removed extension {}", name))),
+        Err(_) => Err(StatusCode::NOT_FOUND),
     }
 }
 
 #[utoipa::path(
-    get,
-    path = "/config",
+    put,
+    path = "/config/extensions/{name}",
+    request_body = ExtensionQuery,
     responses(
-        (status = 200, description = "All configuration values retrieved successfully", body = ConfigResponse)
+        (status = 200, description = "Extension updated successfully", body = String),
+        (status = 404, description = "Extension not found"),
+        (status = 500, description = "Internal server error")
     )
 )]
-pub async fn read_all_config(
+pub async fn update_extension(
     State(state): State<AppState>,
     headers: HeaderMap,
-) -> Result<Json<ConfigResponse>, StatusCode> {
-    // Use the helper function to verify the secret key
+    axum::extract::Path(name): axum::extract::Path<String>,
+    Json(extension_query): Json<ExtensionQuery>,
+) -> Result<Json<String>, StatusCode> {
     verify_secret_key(&headers, &state)?;
 
-    let config = Config::global();
+    let key = name_to_key(&name);
 
-    // Load values from config file
-    let values = config.load_values().unwrap_or_default();
+    // Check if extension exists
+    let extensions = ExtensionManager::get_all().map_err(|_| StatusCode::INTERNAL_SERVER_ERROR)?;
 
-    Ok(Json(ConfigResponse { config: values }))
+    if !extensions.iter().any(|entry| entry.config.key() == key) {
+        return Err(StatusCode::NOT_FOUND);
+    }
+
+    // Use ExtensionManager to update the extension
+    match ExtensionManager::set(ExtensionEntry {
+        enabled: extension_query.enabled,
+        config: extension_query.config,
+    }) {
+        Ok(_) => Ok(Json(format!("Updated extension {}", extension_query.name))),
+        Err(_) => Err(StatusCode::INTERNAL_SERVER_ERROR),
+    }
 }
 
 #[utoipa::path(
-    put,
-    path = "/config/extension",
-    request_body = ExtensionQuery,
+    post,
+    path = "/extensions/{name}/toggle",
     responses(
-        (status = 200, description = "Extension configuration updated successfully", body = String),
+        (status = 200, description = "Extension toggled successfully", body = String),
         (status = 404, description = "Extension not found"),
         (status = 500, description = "Internal server error")
     )
 )]
-pub async fn update_extension(
+pub async fn toggle_extension(
```

Comment:
similar comment as above, can we just use a single set meth to also set enabled/dsiabled? it would make sense to have this if in some cases the UI doesn't have the other components of the extension readily available. but if it does i usually would prefer keeping the surface area smaller